primitives/state-machine/src/basic.rs

```diff
@@ -111,6 +111,8 @@ impl From<BTreeMap<StorageKey, StorageValue>> for BasicExternalities {
 }
 
 impl Externalities for BasicExternalities {
+	fn set_offchain_storage(&mut self, _key: &[u8], _value: Option<&[u8]>) {}
```

Comment:
`place_storage` is using the `StorageValue` and `StorageKey` types. Is there a reason to avoid the same here?

Reply:
None in particular, it was following `Externalities::set_offchain_storage` which also uses `&[u8]`

Reply:
I think in this case it's better to work with borrowed data, cause we can avoid a clone if indexing is disabled. That said we should probably unify the API (and introduce `Storage*Ref` to work with slices). I'd leave it as a separate PR though.